### PR TITLE
#11372 Added methods andHaving(), orHaving(), betweenHaving(), notBetweenHaving(), inHaving(), notInHaving() in `Phalcon\Mvc\Model\Query\Builder`

### PR DESCRIPTION
**Pull request by michanismus**
Hope everything is fine now

### CODE REVIEW

File: phalcon/mvc/model/query/builder.zep [hunk 1]
```diff
@@ -861,15 +861,325 @@ class Builder implements BuilderInterface, InjectionAwareInterface
...
+			}
+			let this->_bindParams = mergedParams;
+		}
+
+		/**
+		 * Merge the bind types to the current ones
+		 */
+		if typeof bindTypes == "array" {
+			let currentBindTypes = this->_bindTypes;
+			if typeof currentBindParams == "array" {
```
Comment (SidRoberts) [inline phalcon/mvc/model/query/builder.zep]: Typo?

Comment (michanismus) [inline phalcon/mvc/model/query/builder.zep]: What exactly?

Comment (SidRoberts) [inline phalcon/mvc/model/query/builder.zep]: Should this be currentBindParams or currentBindTypes?

Comment (michanismus) [inline phalcon/mvc/model/query/builder.zep]: I think so. I just duplicated the where functions. Having function should behave like where so I guess everything is fine.

**Comment by michanismus** [comment]
Any chance to get it merged? :)

**Comment by Jurigag** [comment]
It need to be rebased

**Comment by sergeyklay** [comment]
We have to code freeze the 2.1.x for purposes to testing and preparing for release RC.1. I'll detail consider your PR after release

**Comment by michanismus** [comment]
Closed this PR in favour of https://github.com/phalcon/cphalcon/pull/12769.

**Pull request closed without merging**